src/Calculator/Common/KeyboardShortcuManager.cs

```diff
@@ -0,0 +1,843 @@
+using MUXC = Microsoft.UI.Xaml.Controls;
+
+namespace CalculatorApp
+{
+    namespace Common
+    {
+        static partial class KeyboardShortcutManagerLocals
+        {
+            // Lights up all of the buttons in the given range
+            // The range is defined by a pair of iterators
+            static public void LightUpButtons(IEnumerable<WeakReference> buttons)
+            {
+                foreach (var button in buttons)
+                {
+                    var btn = button.Target as ButtonBase;
+                    if (btn != null && btn.IsEnabled)
+                    {
+                        LightUpButton(btn);
+                    }
+                }
+            }
+
+            static public void LightUpButton(ButtonBase button)
+            {
+                // If the button is a toggle button then we don't need
+                // to change the UI of the button
+                if (button is ToggleButton)
+                {
+                    return;
+                }
+
+                // The button will go into the visual Pressed state with this call
+                VisualStateManager.GoToState(button, "Pressed", true);
+
+                // This timer will fire after lightUpTime and make the button
+                // go back to the normal state.
+                // This timer will only fire once after which it will be destroyed
+                var timer = new DispatcherTimer();
+                TimeSpan lightUpTime = TimeSpan.FromMilliseconds(500); // half second
+                timer.Interval = lightUpTime;
+
+                var timerWeakReference = new WeakReference(timer);
+                var buttonWeakReference = new WeakReference(button);
+                timer.Tick += (sender, args) =>
+                {
+                    var btn = buttonWeakReference.Target as ButtonBase;
+                    if (btn != null)
+                    {
+                        VisualStateManager.GoToState(button, "Normal", true);
+                    }
+
+                    var tmr = timerWeakReference.Target as DispatcherTimer;
+                    if (tmr != null)
+                    {
+                        tmr.Stop();
+                    }
+                };
+                timer.Start();
+            }
+
+            // Looks for the first button reference that it can resolve
+            // and execute its command.
+            // NOTE: It is assumed that all buttons associated with a particular
+            // key have the same command
+            static public void RunFirstEnabledButtonCommand(IEnumerable<WeakReference> buttons)
+            {
+                foreach (var button in buttons)
+                {
+                    var btn = button.Target as ButtonBase;
+                    if (btn != null && btn.IsEnabled)
+                    {
+                        RunButtonCommand(btn);
+                        break;
+                    }
+                }
+            }
+
```

Comment:
There are many places have too many empty lines, some have 2 lines and some have 3 lines. Could you please reformat that?

Reply:
done